tests/ui/models/filter_alerts_with_videos_test.js

```diff
@@ -0,0 +1,103 @@
+
+const dummyFrameworkName = 'browsertime';
+
+describe('FilterAlertsWithVideos', () => {
+  afterEach(() => {
+    fetchMock.reset();
+  });
+
+  describe('File bug url when alerts contain browsertime results links', () => {
+    beforeEach(() => {
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=847217`,
+        currentJoblistWithVideoResultsOne,
+      );
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=847217&page=2`,
+        currentJoblistWithVideoResultsTwo,
+      );
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=846998`,
+        prevJoblistWithVideoResultsOne,
+      );
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=846998&page=2`,
+        prevJoblistWithVideoResultsTwo,
+      );
+    });
+
+    test('should return alerts with browsertime results links', async () => {
+      const alertsWithVideos = new FilterAlertsWithVideos(
+        testAlertSummaryWithVideos,
+        [{ id: 13, name: dummyFrameworkName }],
+      );
+      const expectedResultsLink =
+        'https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/c6nAPFniThOiaGvDT2DNCw/runs/0/artifacts/public/test_info/browsertime-results.tgz';
+      const expectedPrevResultsLink =
+        'https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/LUjQR22tRp6J4wXLHhYk8g/runs/0/artifacts/public/test_info/browsertime-results.tgz';
+
+      const alerts = await alertsWithVideos.getFilteredAlerts();
+      alerts.forEach((alert) => {
+        if (alertsWithVideos.shouldHaveVideoLinks(alert)) {
+          expect(alert).toHaveProperty('results_link', expectedResultsLink);
+          expect(alert).toHaveProperty(
+            'prev_results_link',
+            expectedPrevResultsLink,
+          );
+        } else {
+          expect(alert.results_link).toBeUndefined();
+          expect(alert.prev_results_link).toBeUndefined();
+        }
+      });
+    });
+  });
+
+  describe('File bug url when alerts doesnt contain browsertime results links', () => {
+    beforeEach(() => {
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=851190`,
+        currentJoblistWithoutVideoResultsOne,
+      );
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=851190&page=2`,
+        currentJoblistWithoutVideoResultsTwo,
+      );
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=847398`,
+        prevJoblistWithoutVideoResultsOne,
+      );
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=847398&page=2`,
+        prevJoblistWithoutVideoResultsTwo,
+      );
+    });
+
+    test('should return alerts without browsertime results links', async () => {
+      const alertsWithoutVideos = new FilterAlertsWithVideos(
+        testAlertSummaryWithoutVideos,
+        [{ id: 13, name: dummyFrameworkName }],
+      );
+
+      const alerts = await alertsWithoutVideos.getFilteredAlerts();
```

Comment:
We can assert that we didn't get any filtered alert, then iterate through **all** alerts & check they weren't  enriched.

Reply:
getFilteredAlerts returns the already enriched alerts and only that ones. If there's no alert with videos, it won't return anything, so probably checking that alerts is [] should be enough.